functions/api/jisho.py

```diff
@@ -38,83 +42,103 @@ async def run_error(self, ctx, error):
                 senses['parts_of_speech'].remove('Wikipedia definition')
             except ValueError:
                 pass
-
-            result = {'readings': list(
-                readings), 'words': list(words), 'kanji': list(kanji_link), **senses}
+            
+            result = {
+                'readings': list(readings),
+                'words': list(words),
+                'kanji': list(kanji_link),
+                **senses
+            }
+            
             results.append(result)
+        
         return self.jisho_embed(keyword, results)
-
+    
     def jisho_embed(self, keywords, stats_json):
-        results = []
         if not stats_json:
             return None
-
+        
+        results = []
+        
         size = len(stats_json)
+        color = self.bot.color
+        jisho_url = self.jisho_url.format("%20".join(keywords.split()))
+        
         for start, res in enumerate(stats_json, 1):
-            res = {key: "\n".join(val) or "None" for key,
-                   val in res.items()}
+            res = {
+                key: "\n".join(val) or "None" for key, val in res.items()
+            }
             res['english'] = ", ".join(res['english'].split("\n"))
-            em = {
+            
+            embed_dict = {
                 "title": keywords,
-                "color": self.bot.color,
-                "url": self.jisho_url.format("%20".join(keywords.split())),
-                "fields": [{
-                    "name": "Words",
-                    "value": res['words']
-                }, {
-                    "name": "Kanji",
-                    "value": res['kanji']
-                }, {
-                    "name": "Readings",
-                    "value": res['readings']
-                }, {
-                    "name": "Parts of Speech",
-                    "value": res['parts_of_speech']
-                }, {
-                    "name": "Meanings",
-                    "value": res['english']
-                }],
+                "color": color,
+                "url"  : jisho_url,
+                "fields": [
+                    {
+                        "name": "Words",
+                        "value": res['words']
+                    }, {
+                        "name": "Kanji",
+                        "value": res['kanji']
+                    }, {
+                        "name": "Readings",
+                        "value": res['readings']
+                    }, {
+                        "name": "Parts of Speech",
+                        "value": res['parts_of_speech']
+                    }, {
+                        "name": "Meanings",
+                        "value": res['english']
+                    }
+                ],
                 "footer": {"text": f"Page {start}/{size}"}
             }
-            embed = Embed.from_dict(em)
-            results.append(embed)
+            
+            results.append(Embed.from_dict(embed_dict))
```

Comment:
So... You're separating values into variables on line 103. But here you're doing the opposite? Why? Once again, I don't get it.

Reply:
Agreed and nesting function calls with other function calls is not always good practice from a readability stand point